pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -925,10 +925,22 @@ self: super: {
     unordered-containers = self.unordered-containers_0_2_5_1;
     cpphs = self.cpphs_1_19_3;
   };
-  Agda = super.Agda.override {
+  Agda = overrideCabal (super.Agda.override {
     unordered-containers = self.unordered-containers_0_2_5_1;
     cpphs = self.cpphs_1_19_3;
-  };
+  }) (drv: {
+    postInstall = ''
+      # Separate loops to avoid internal error
+      files=($out/share/*-ghc-*/Agda-*/lib/prim/Agda/{Primitive.agda,Builtin/*.agda})
+      for f in "''${files[@]}" ; do
+        $out/bin/agda $f
+      done
+      for f in "''${files[@]}" ; do
+        $out/bin/agda -c --no-main $f
+      done
+      $out/bin/agda-mode compile
+    '';
+  });
 
```

Comment:
It feels a little weird to have extra code in `cabal2nix` that configures a [`postInstall` hook](https://github.com/NixOS/cabal2nix/blob/master/distribution-nixpkgs/src/Distribution/Nixpkgs/Haskell/FromCabal/PostProcess.hs#L199-L205) when, apparently, that code does not work and needs to be overridden yet again. This is going to very confusing for our users.